core/node/groups.go

```diff
@@ -77,6 +77,7 @@ func LibP2P(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 		pubsubOptions = append(
 			pubsubOptions,
 			pubsub.WithMessageSigning(!cfg.Pubsub.DisableSigning),
+			pubsub.WithSeenMessagesTTL(cfg.Pubsub.SeenMessagesTTL.WithDefault(pubsub.TimeCacheDuration)),
```

Comment:
Hey @lidel, it took me a bit to get everything figured out and setup with `sharness`. After playing around with it a bit, I'm realizing that there isn't really a good way to induce duplicate pubsub messages. The API (rightly) doesn't allow setting a message ID explicitly.
```
func (t *Topic) Publish(ctx context.Context, data []byte, opts ...PubOpt) error {
.
.
.
	return t.p.val.PushLocal(&Message{m, "", t.p.host.ID(), nil, pub.local})
}
```
The only option I can currently come up with is to use `tc netem` via command line or [code](https://github.com/florianl/go-tc) to simulate network traffic duplication.

Does this seems like a viable approach? Not sure what the restrictions are around what tests are/aren't allowed to do. I've used `tc netem` before so I can figure this out, if acceptable.

Reply:
I'm thinking aloud here but I could also make a `libp2p-sim` tool under `test/dependencies` that connects to libp2p like a normal node but is capable of sending spurious messages. Might be useful in other contexts too perhaps.

Reply:
> probably not, we do go out of our way to supress them.

Ok, understood, @vyzo, thanks for chiming in 🙏🏼

@lidel, in case testing this isn't possible/straightforward, what would be the best way to proceed here?

Reply:
If worth experimenting with (I'm not an expert on `go-libp2p-pubsub` so will have to play around with it), I can try to get the `libp2p-sim` in soon-ish.

I'm also working on fixing the other cache-related [bug](https://github.com/libp2p/go-libp2p-pubsub/issues/502), and will then try to have that included in `go-ipfs` as well, so some kind of cache-related test/framework might be beneficial for that too.

Though, if this all feels like not enough ROI, maybe we don't add cache-specific tests? Will do whatever y'all recommend.

Reply:
> What do you think about having a libp2p-sim that implements it's own Publish and can send dups?

This and https://github.com/libp2p/go-libp2p-pubsub/issues/502 both sound sensible :+1:

Reply:
@lidel, would it be possible to do a regular Golang unit test here instead of adding to `sharness`?

Reply:
@lidel, after a lot of hair pulling, I was unable to set up a libp2p simulator but I was able to figure out how to write a Go test under `test/integration` that does the job, IMO. Would it work to keep the test there? Let me know what you think.

FYI, this test uncovered another pubsub/cache [bug](https://github.com/ipfs/kubo/pull/9372/files#diff-a75543978c37023721256291000855b02598e5d72668e203f7372cc4b8c159cfR222), which I will fix under [this](https://github.com/libp2p/go-libp2p-pubsub/issues/502) issue.

Reply:
Thank you for doing due diligence on this, yes, your approach is acceptable – will act as an early warning if go-libp2p APIs change, or a regression is introduced.

Reply:
> Thank you for driving this @smrz2001, and being patient with test work. I'm merging this, so we can include it in Kubo 0.18.0-rc1 (ETA second week of December, or beginning of January).

Excellent, thanks so much, @lidel!! 🙏🏼🙏🏼

I'll try to get the other bugs fixed soon as well, and will update the dependencies and tests accordingly.